tests/e2e/e2e.sh

```diff
@@ -17,83 +17,54 @@
-    exit 1
-  fi
-
-  KUBECONFIG="$(kind get kubeconfig-path --name="istio-testing")"
-  export KUBECONFIG
-
-cat <<EOF > metallb-config.yaml
-apiVersion: v1
-kind: ConfigMap
-metadata:
-  namespace: metallb-system
-  name: config
-data:
-  config: |
-    address-pools:
-    - name: default
-      protocol: layer2
-      addresses:
-      - 172.17.255.1-172.17.255.250
-EOF
-
-  kubectl apply -f https://raw.githubusercontent.com/google/metallb/v0.8.1/manifests/metallb.yaml
-  kubectl apply -f ./metallb-config.yaml
-}
 
 function setup_docker() {
-  HUB=istio-testing TAG=latest make -f Makefile.core.mk controller docker
-  kind --loglevel debug --name istio-testing load docker-image istio-testing/operator:istio-testing
+  HUB=gcr.io/istio-testing TAG=1.5-dev make controller docker
+  kind --loglevel debug --name istio-testing load docker-image gcr.io/istio-testing/operator:1.5-dev
+  kind --loglevel debug --name istio-testing load docker-image istio-testing/app:istio-testing
 }
 
-setup_kind_cluster
-setup_docker
+
 
 mkdir -p "${ARTIFACTS}/out"
 
-ISTIO_DIR="${GOPATH}/src/istio.io/istio"
+ISTIO_DIR="${ROOT}/src/istio.io/istio"
 
 # Create a clone of the Istio repository
 if [[ ! -d "${ISTIO_DIR}" ]]
 then
-  git clone https://github.com/sdake/istio.git "${ISTIO_DIR}"
+  git clone https://github.com/istio/istio.git "${ISTIO_DIR}"
 fi
 
 # Create an operator manifest from the default control plane configuration
-
-operator_manifest_files=( "deploy/namespace.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_crd.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_cr.yaml" "deploy/service_account.yaml" "deploy/clusterrole.yaml" "deploy/clusterrole_binding.yaml" "deploy/service.yaml" "deploy/operator.yaml" )
+cd "${ROOT}/src/istio.io/operator"
+operator_manifest_files=( "deploy/namespace.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_crd.yaml" "deploy/service_account.yaml" "deploy/clusterrole.yaml" "deploy/clusterrole_binding.yaml" "deploy/service.yaml" "deploy/operator.yaml" "deploy/crds/istio_v1alpha2_istiocontrolplane_cr.yaml" )
 
 # Generate the main manifest
 rm -f "${ISTIO_DIR}"/install/kubernetes/istio-operator.yaml
 for manifest_file in "${operator_manifest_files[@]}"
 do
 	cat "${manifest_file}" >> "${ISTIO_DIR}"/install/kubernetes/istio-operator.yaml
-	echo "---" >> "${ISTIO_DIR}"/install/kubernetes/istio-operator.yaml
 done
 
-kubectl get pods --all-namespaces -o wide
+
+#kind cluster setup
+pushd "${ISTIO_DIR}"
+# shellcheck disable=SC1091
+source "./prow/lib.sh"
+setup_kind_cluster ""
+KUBECONFIG=$(kind get kubeconfig-path --name="istio-testing")
+export KUBECONFIG
+HUB="${HUB}" TAG="${TAG}" make docker.app
+popd
+
+setup_docker
 
 pushd "${ISTIO_DIR}" || exit
   make istioctl
-
-  HUB=gcr.io/istio-testing TAG=latest E2E_ARGS="--use_operator --test_logs_path=${ARTIFACTS}" make e2e_simple_run
+  HUB="${HUB}" TAG="${TAG}" E2E_ARGS="--use_operator --use_local_cluster=true --test_logs_path=${ARTIFACTS}" make e2e_simple_noauth_run
```

Comment:
nit - we should probably using gcr for the hub for all Istio components, and then the local image for the operator. We aren't testing istio/istio here so no need to build those images

Reply:
just need to build istio-testing/app:istio-testing as we will not build this images during normal istio build

Reply:
I think We should not be building it, we should use gcr.io/istio-testing/app:latest. We aren't building pilot or galley, so why build the test app?

Reply:
> the istioctl inject is not related to this pr, I have a PR on istio/istio to fix it

Thanks for the info!

Reply:
changed to use gcr.io/istio-testing/app:latest and skip building istio-testing/app